Implement RAC Style Library

HaSistrunk opened the pull request:
Fix #208 

Implements the RAC Style Library v0.12.0 and makes some significant design changes to the site homepage.
- Replaces the card layout used on the previous site homepage with a searchable table.
- Removes docs categories and category filter script. These categories were no longer very accurate or useful to find documentation since the scope and number of docs has grown significantly since the initial site design.
- Uses typography and layout from rockarch.org website design for text-heavy pages and table of contents. For this change, we'll want to update the docs config files and user guide on how to add new docs to remove categories.
- Updates jquery version (which is still be used mainly for datatables.js)
- Adds backstopjs for future visual regression testing when we make changes to the site.

## Details of work in 3 sprints

### Sprint 1
1. Removed links to GitHub in the header nav that we thought were unnecessary
2. Implemented header styles (let's chat about use of the RAC logo on our sites overall - do we want to use it in the header?)
3. Implemented card styles on the homepage including badges for "internal only" links. To see what this looks like in the UI, you have to change line 14 in `index.html` to `{% if doc.public == true %}` and give the site time to rebuild, because the local build doesn't include internal sites.
4. Implemented table of contents, including adding javascript to `toc.js` that adds an "active" css class to the toc links when the corresponding header (h2) for that link is in view on the page while scrolling.
5. Added a few layout details for the table of contents and centered page titles

### Sprint 2
- [x] 1. Implement footer styles
- [x] 2. Implement styles for the "filter by category" feature on the homepage (wondering if these should actually be coded as [radio buttons](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio) to follow known patterns and be more accessible)
- [x] 3. Implement skip links for main page
- [x] 4. Implement skip links into table of contents nav
- [x] 5. Review styles for codeblocks, and `.docs-example` (tweaked .docs-example background to be a Style Library color, but otherwise no changes).
- [x] 6. Make tables responsive (at least scrollable) when they are too large and go off the screen
- [x] 7. Fix layout. Look specifically at bottom margins for headings, tables, code blocks, and the `docs-example` classes to make sure everything works well in different documentation.
- [x] 8. Review all styles and HTML for unused classes and delete them
- [x] 9. Review necessity of Bootstrap JS and other javascript libraries - removed popper.js, list.js and bootstrap.js
- [x] 10. Try a sortable data table on the homepage instead of using cards. 
- [x] 11. This site is very text heavy - make body text a little larger and use relative units instead of pixels for fonts. Also create styles that don't exist for h4, h5, and h6. These styles should be added back into the Style Library.

### Sprint 3
Stepping back and considering some larger design questions and actions:
- [x] Improve documentation titles overall. Remove "Rockefeller Archive Center" from docs titles, create guidance around descriptive titles so that not everything is a "guide to...", for example. Consider any technical implications of titles changing. **Solution:** We're working on this outside of this PR.
- [x] In the absence of a search function, what is the best way to help users find the documentation that they are looking for/might be interested in? Reconsider current pattern of category filtering and the categories themselves. **Solution:** remove categories and replace with docs title search function on the homepage using datatables.js.
- [x]  Improve readability for this text heavy site. We don't have designers from designers focused on this kind of text-heavy site (this applies to the blog as well). Research best practices, but also consider re-engaging designers to help us identify the best approach. **Solution**: We implemented layout and typography styles from rockarch.org on text-heavy pages.
- [x] Table of contents active link styles seem unfinished. Improve them, or engage designers to improve them. **Solution:** improved designs ourselves.

HaSistrunk committed:
import rac style library + remove bootstrap styles

HaSistrunk committed:
start to implement header styles

HaSistrunk committed:
implement table of contents

kemartin2015 committed:
implement card styles

kemartin2015 committed:
change number of card columns, add badge class

HaSistrunk committed:
center h1s on the page

HaSistrunk committed:
remove category.html include and make badge a div

HaSistrunk committed:
adjust card layout

HaSistrunk committed:
better align logo with brand text in header

HaSistrunk committed:
implement toc mobile layout and add toc active styles

HaSistrunk committed:
add toc.js function to highlight active toc link while scrolling page

HaSistrunk committed:
improve custom styles documentation

helrond commented on theme/_layouts/default.html:
```diff
@@ -9,17 +8,17 @@
-       <div class="container mt-5">
-         <h1 class="display-5 text-center">{{ page.title }}</h1>
+       <!-- <div class="container mt-5"> -->
+         <h1 class="wrapper">{{ page.title }}</h1>
```
Why is there a `.wrapper` class on the `h1` element element?

HaSistrunk replied:
It was a lazy way to center the text on the page...

helrond commented on theme/_layouts/docs.html:
```diff
@@ -2,17 +2,15 @@
 
-<div class="container mt-4">
-  <h1 class="display-5 text-center">{{ page.title }}</h1>
+<div class="wrapper">
```
is this wrapping div necessary?

helrond commented on theme/js/toc.js:
```diff
@@ -74,5 +74,35 @@
 
     render[settings.showEffect]();
+
+    var tocLinkItems = $('#current > a.toc__link-item'); // Get all the TOC header links
```
I would consider putting this in a different JS file because it's a very different function than the link creation, which is what the rest of this file is doing.

helrond replied:
That would also potentially allow us to progressively start shedding jQuery where we can.

helrond commented on theme/js/toc.js:
```diff
@@ -74,5 +74,35 @@
+    var tocLinkItems = $('#current > a.toc__link-item'); // Get all the TOC header links
+    var activeLink = null; // Store the currently active TOC link
+
+    $(window).on('scroll', function() {
```
I think this will need to handle scroll up and down differently. For example, if I go to /archivematica-local/appendix/ and scroll all the way down, the last header gets highlighted. Then, as I scroll up and that header and the content under it is not longer visible, that header is still highlighted until the previous header is visible. Given that the content I am seeing is from that previous header it's a little confusing...hope that makes sense

helrond replied:
Another thing I'm seeing with this script is that if a very short section exists at the end of page, that header will never be highlighted because the header never gets far enough up the page. This could maybe be addressed with bottom padding or something. Examples:
    - http://localhost:4000/Archival-Education-Strategy-Document/ 
    - http://localhost:4000/archivematica-local/transferAndSetup/

helrond reviewed (commented):
This is looking good! I've added a few comments on specific things below but a few global comments:
- Have we thought at all about revising the home page layout? Given the number of things here, I'm personally finding the card layout somewhat inefficient and hard to navigate, even when I know what I'm looking for.
- I would personally vote no logo in the header, but instead use it in the homepage hero as we have on DIMES, rockarch.org and the library website.

HaSistrunk commented:
The team agreed to remove the logo from the header, which is something we will also remove from the Style Guide as an option.

HaSistrunk committed:
remove logo in header

HaSistrunk committed:
remove boostrapjs

HaSistrunk committed:
Use flex and grid to apply global layout, remove home layout, center h1s

kemartin2015 committed:
implement footer style, add skip-links to nav and main page

kemartin2015 committed:
fix skip links styling

HaSistrunk committed:
move active headers action to its own script file

HaSistrunk committed:
remove unnecessary ids and classes from toc

HaSistrunk committed:
add functionality to make last link active at bottom of page

HaSistrunk committed:
Use more performant active toc script that handles scrolling up better

HaSistrunk committed:
inherit font size for links inside headings in docs

HaSistrunk committed:
Add radio group semantics and styles (not function)

HaSistrunk committed:
add category filter functionality

HaSistrunk committed:
cleanup radio group layout styles

HaSistrunk committed:
improve category filter accessibility

HaSistrunk committed:
simplify live region

HaSistrunk committed:
remove unused scripts

HaSistrunk commented:
> Have we thought at all about revising the home page layout? Given the number of things here, I'm personally finding the card layout somewhat inefficient and hard to navigate, even when I know what I'm looking for.

I'm wondering what folks think about something like this that presents the docs in a table. We'd have to think about simplifying the info on mobile, maybe remove the last updated dates on small screens or something similar to make it responsive.
<img width="1416" alt="image" src="https://github.com/RockefellerArchiveCenter/docs-build/assets/15254041/deac9ff4-3ec5-4cbb-8d6d-12a34ddc7b2d">

We could order the table in different ways or get rid of the filters and add page headers/sections that correspond to each category. The problem with that is some docs are in multiple categories.

HaSistrunk committed:
differentiate toc focus and active styles

kemartin2015 committed:
Removes commented-out code

kemartin2015 committed:
make tables responsive

helrond commented:
I'm liking the table layout, particularly if we could make it sortable.

If I remember correctly, the addition of categories was really a way to implement some limited search capability without adding a full index and keyword search to the site. I'm wondering if we need to revisit that thinking too, because those categories feel like they don't totally fit the current set of documentation.

It also occurred to me that another think that might be very helpful on the home page is a brief (like one-sentence) description of what the documentation is. It's not always obvious from the title, particularly for folks who don't already know what they're looking for.

HaSistrunk committed:
adjust typography for font size and improve element margins

HaSistrunk committed:
Use sortable table instead of cards

HaSistrunk commented:
> I'm liking the table layout, particularly if we could make it sortable.

Ok! I've added a sortable table on the homepage that replaces the cards. Let's see what we think. I didn't fix the category filter functionality for the new table approach, because I want to see what direction we're going in before investing the time into that.

> If I remember correctly, the addition of categories was really a way to implement some limited search capability without adding a full index and keyword search to the site. I'm wondering if we need to revisit that thinking too, because those categories feel like they don't totally fit the current set of documentation.

Yes, this is correct. We did a card sort with staff with the documentation we had then, and these are the categories that emerged. @kemartin2015 was also saying that the categories now seem out of date. I still think some kind of filtering is useful, though, with so much documentation. We have data designating each as a policy/workflow and these categories (plus a description that we use in the title and helps SEO). We could use policy/workflow for a basic sort, or go back to the drawing board on categories. Maybe another card sort could be useful here?

> It also occurred to me that another think that might be very helpful on the home page is a brief (like one-sentence) description of what the documentation is. It's not always obvious from the title, particularly for folks who don't already know what they're looking for.

I added one based on how the site is described on rockarch.org - open to tweaks/feedback on that language!

HaSistrunk committed:
use style library favicon

HaSistrunk committed:
add margin to superscript footnotes

HaSistrunk committed:
copy rockarch.org layout, improve toc active styles

HaSistrunk committed:
heading style tweaks

HaSistrunk committed:
update jquery

HaSistrunk committed:
redesign homepage with table search + organize custom styles

HaSistrunk committed:
table styles tweaks

HaSistrunk committed:
tweak homepage styles and lint toc id

HaSistrunk committed:
use the correct reading room hours in footer

HaSistrunk committed:
implement backstopjs regression testing

HaSistrunk committed:
include installing BackstopJS dependency

helrond reviewed (approved):
I'm into this!

HaSistrunk committed:
update RAC style library to version 0.11.0

HaSistrunk committed:
fix mobile toc layout and remove active styles

HaSistrunk committed:
add margin to badge in table, update backstop ref imgs

HaSistrunk committed:
remove categories and tags from config in readme

HaSistrunk committed:
fix header layout on mobile

HaSistrunk committed:
Merge branch 'development' into issue-208
i the commit.
merge development into issue-208

HaSistrunk committed:
update backstop ref imgs

HaSistrunk committed:
update social media icons in footer

HaSistrunk committed:
update backstop reference imgs
